wrangler-storage/src/main/java/io/cdap/wrangler/store/recipe/RecipeStore.java

```diff
@@ -89,37 +89,61 @@ public RecipeStore(TransactionRunner transactionRunner) {
    * @param recipeRow recipe to create/update
    */
   public void saveRecipe(RecipeId recipeId, RecipeRow recipeRow) {
-    saveRecipe(recipeId, recipeRow, false);
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable table = context.getTable(TABLE_ID);
+
+      // check if a different recipe with same name exists
+      String recipeName = recipeRow.getRecipe().getRecipeName();
+      RecipeRow oldRecipeWithName = getRecipeByName(table, recipeName, recipeId.getNamespace(), false);
+      if (oldRecipeWithName != null) {
+        throw new RecipeAlreadyExistsException(String.format("recipe with name '%s' already exists", recipeName));
+      }
+      saveRecipe(recipeId, recipeRow, false, table);
+    });
   }
 
-  // Create or update a recipe
-  private void saveRecipe(RecipeId recipeId, RecipeRow recipeRow, boolean failIfNotFound)
-    throws RecipeAlreadyExistsException {
+  /**
+   * Update a recipe
+   * @param recipeId  id of the recipe
+   * @param recipeRow recipe to create/update
+   */
+  public void updateRecipe(RecipeId recipeId, RecipeRow recipeRow) {
     TransactionRunners.run(transactionRunner, context -> {
       StructuredTable table = context.getTable(TABLE_ID);
 
+      // check if a different recipe with same name exists
       String recipeName = recipeRow.getRecipe().getRecipeName();
-      RecipeRow oldRecipeWithName = getRecipeByName(table, recipeName, recipeId.getNamespace(), failIfNotFound);
-      if (!failIfNotFound && oldRecipeWithName != null) {
+      RecipeRow oldRecipeWithName = getRecipeByName(table, recipeName, recipeId.getNamespace(), false);
+      if (oldRecipeWithName != null
+        && !oldRecipeWithName.getRecipe().getRecipeId().equals(recipeRow.getRecipe().getRecipeId())) {
         throw new RecipeAlreadyExistsException(String.format("recipe with name '%s' already exists", recipeName));
       }
+      saveRecipe(recipeId, recipeRow, true, table);
+    });
+  }
 
-      RecipeRow oldRecipe = getRecipeInternal(table, recipeId, failIfNotFound);
-      RecipeRow newRecipe = recipeRow;
-      if (oldRecipe != null) {
-        Recipe updatedRecipe = Recipe.builder(newRecipe.getRecipe())
-          .setCreatedTimeMillis(oldRecipe.getRecipe().getCreatedTimeMillis()).build();
-        newRecipe = RecipeRow.builder(updatedRecipe).build();
-      }
+  // Create or update a recipe
+  // failIfNotFound should be false for create, and true for update
+  private void saveRecipe(RecipeId recipeId, RecipeRow recipeRow, boolean failIfNotFound, StructuredTable table)
+    throws IOException, RecipeAlreadyExistsException {
+    RecipeRow oldRecipe = getRecipeInternal(table, recipeId, failIfNotFound);
+    if (oldRecipe != null) {
+      Recipe updatedRecipe = Recipe.builder(recipeRow.getRecipe())
+        .setCreatedTimeMillis(oldRecipe.getRecipe().getCreatedTimeMillis()).build();
```

Comment:
create timestamp is already available in oldRecipeWithName in updateRecipe, should we move this logic there? It is needed only for update. We may need similar value-fill for columns that are not passed from UI for update.